### ISSUE TITLE
Different outputs between using input_embeds and using input_ids

### ISSUE THREAD

**Issue by HCRTY** [huggingface/transformers#40098]
I trained qwen2.5 with a TTS task. When I do inference, I tried 2 different input type: input_ids and input_embeds. However, I got different outputs from the same model. I wonder whether this is expected?

**Comment by JawadAliAI** [comment]
Hi, I’ve also encountered this issue while fine-tuning Qwen2.5 for a TTS task. I noticed the same behavior: input_ids and inputs_embeds give different outputs unless positional embeddings and attention masks are explicitly aligned.

I’d like to work on this and help investigate/fix it. Could you please assign this issue to me, or let me know if there are any specific guidelines I should follow to contribute a PR?

Thanks!

**Comment by Rocketknight1** [comment]
We don't do issue assignment, but if you think there's a bug feel free to submit a PR!
